editor-components/stats/stats2/stats2.tsx

```diff
@@ -177,38 +130,113 @@ class Stats2Page extends BaseStats {
-              )
-            )}
+          <div className={this.decorateCSS("header-wrapper")}>
+            <div className={this.decorateCSS("header")}>{this.getPropValue("header")}</div>
+            <div className={this.decorateCSS("subHeader")}>
+              {this.getPropValue("subHeader")}
+              <button className={this.decorateCSS("contact-button")}>{this.getPropValue("contactButton")}</button>
+            </div>
+          </div>
+
+          <div className={this.decorateCSS("cards-container")}>
+            {cards.map((card, index) => (
+              <AnimatedCard key={index} card={card} styles={styles} />
+            ))}
           </div>
         </div>
       </div>
     );
   }
 }
 
+type AnimatedCardProps = {
+  card: Card;
+  styles: typeof styles;
+};
+
+type AnimatedCardState = {
+  digits: number[];
+};
+
+class AnimatedCard extends React.Component<AnimatedCardProps, AnimatedCardState> {
+  private observer: IntersectionObserver;
+  private ref: React.RefObject<HTMLDivElement>;
+
+  constructor(props: AnimatedCardProps) {
+    super(props);
+    const initialDigits = Array.from(String(props.card.amount), () => 0);
+    this.state = {
+      digits: initialDigits,
+    };
+    this.ref = React.createRef();
+  }
+
+  componentDidMount() {
+    this.observer = new IntersectionObserver(
+      (entries) => {
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            this.animateDigits();
+            this.observer.unobserve(entry.target);
+          }
+        });
+      },
+      { threshold: 0.5 }
+    );
+
+    if (this.ref.current) {
+      this.observer.observe(this.ref.current);
+    }
+  }
+
+  animateDigits() {
+    const { digits } = this.state;
+    const finalDigits = Array.from(String(this.props.card.amount), (digit) => parseInt(digit, 10));
+
+    finalDigits.forEach((finalDigit, index) => {
+      if (finalDigit !== digits[index]) {
+        this.animateDigit(index, finalDigit - 1);
+      }
+    });
+  }
+
+  animateDigit(index: number, finalValue: number) {
+    let currentValue = 0;
+    const increment = () => {
+      if (currentValue <= finalValue) {
+        this.setState((prevState) => {
+          const newDigits = [...prevState.digits];
+          newDigits[index] = currentValue;
+          return { digits: newDigits };
+        });
+        currentValue += 1;
+        if (currentValue <= finalValue) {
+          setTimeout(increment, 60);
+        }
+      }
+    };
+    increment();
+  }
+
+  render() {
+    const { digits } = this.state;
+
+    return (
+      <div ref={this.ref} className={this.props.styles["listed"]}>
+        <p className={this.props.styles["card-text"]}>{this.props.card.text}</p>
+        <p className={this.props.styles["card-amount"]}>
+          {digits.map((digit, index) => (
+            <span key={index} className={digit > 0 ? this.props.styles["digit"] : ""}>{digit}</span>
+          ))}
+        </p>
+      </div>
+    );
+  }
+}
```

Comment:
you do not need to create another component for this. there is an example for number animation in stats6.